db/migrate/20230428215747_create_collaborations.rb

```diff
@@ -0,0 +1,11 @@
+class CreateCollaborations < ActiveRecord::Migration[7.0]
+  def change
+    create_table :collaborations do |t|
+      t.integer :relationship_type
+      t.string :stakeholder_type
```

Comment:
It's best to add these as references (as it also indexes them):

```suggestion
      t.references :stakeholder, polymorphic: true
```